File: src/Evaluate Reverse Polish Notation.cpp
#include <vector>
#include <iostream>
#include <stack>
#include <string>
#include <sstream>
using namespace std;

class Solution {
public:
	int evalRPN(vector<string> &tokens) {
		stack<string> st;
		for(int i=0; i<tokens.size(); i++)
		{
			if(tokens[i] == "+" || tokens[i] == "-" || tokens[i] == "*" || tokens[i] == "/")
			{
				int a = atoi(st.top().c_str());
				st.pop();
				int b = atoi(st.top().c_str());
				st.pop();
				int num = 0;
				if(tokens[i] == "+")
					num = a + b;
				else if(tokens[i] == "-")
					num = b - a;
				else if(tokens[i] == "*")
					num = b * a;
				else if(tokens[i] == "/")
					num = b / a;
				ostringstream oss;
				oss << num;
				st.push(oss.str());
			}
			else
			{
				st.push(tokens[i]);	
			}			
		}
		return atoi(st.top().c_str());
	}
};

int main()
{

	return 0;
}

File: python/Evaluate Reverse Polish Notation.py
class Solution:
    # @param tokens, a list of string
    # @return an integer
    def evalRPN(self, tokens):
        li = []
        for s in tokens:
            if s == "+" or s == "-" or s == "*" or s == "/":
                a = int(li.pop())
                b = int(li.pop())
                #print("a=%d, b=%d" % (a, b))
                if s == "+":
                    num = b + a
                elif s == "-":
                    num = b - a
                elif s == "*":
                    num = b * a
                elif s == "/":
                    num = int(b / float(a))
                li.append(str(num))
            else:
                li.append(s)
            #print(li)
        return int(li.pop())

if __name__ == '__main__':
    so = Solution()
    tokens = ["10","6","9","3","+","-11","*","/","*","17","+","5","+"]
    print(so.evalRPN(tokens))